unit_test/sparse/Test_Sparse_spgemm.hpp

```diff
@@ -285,16 +280,17 @@ void test_spgemm(lno_t numRows, size_type nnz, lno_t bandwidth, lno_t row_size_v
   lno_t numCols = numRows;
   crsMat_t input_mat = KokkosKernels::Impl::kk_generate_sparse_matrix<crsMat_t>(numRows,numCols,nnz,row_size_variance, bandwidth);
 
-
   crsMat_t output_mat2;
   run_spgemm<crsMat_t, device>(input_mat, input_mat, SPGEMM_DEBUG, output_mat2);
 
-  SPGEMMAlgorithm algorithms [] = {SPGEMM_KK_MEMORY, SPGEMM_KK_SPEED, SPGEMM_KK_MEMSPEED, /*SPGEMM_CUSPARSE, */SPGEMM_MKL};
-
-  for (int ii = 0; ii < 4; ++ii){
+  std::vector<SPGEMMAlgorithm> algorithms = {SPGEMM_KK_MEMORY, SPGEMM_KK_SPEED, SPGEMM_KK_MEMSPEED};
 
-    SPGEMMAlgorithm spgemm_algorithm = algorithms[ii];
+#ifdef HAVE_KOKKOSKERNELS_MKL
```

Comment:
KOKKOSKERNELS_ENABLE_TPL_MKL is not needed ? I thought that is needed for some Trilinos path.

Reply:
@srajama1 Actually I need to check that. ``HAVE_KOKKOSKERNELS_MKL`` is the standard Tribits macro to use. Both that and ``KOKKOSKERNELS_ENABLE_TPL_MKL`` appear in the config.h.in but I'm not sure what the difference is.

Reply:
@jjwilke Is ``KOKKOSKERNELS_ENABLE_TPL_MKL`` the macro to use? Grepping for ``HAVE_KOKKOSKERNELS_MKL`` only shows up in this SPGEMM test and the config.h.in, so is it just a leftover thing from when we used the Tribits name conventions?